Rails の i18n を理解する の課題を作成しました

a-kuroki-gs opened the pull request:
日本語と英語の対応を行う

a-kuroki-gs committed:
Rails の i18n を理解する の課題を作成しました

日本語と英語の対応を行う

yuuu commented on app/controllers/books_controller.rb:
```diff
@@ -25,7 +25,7 @@ def create
 
     respond_to do |format|
       if @book.save
-        format.html { redirect_to book_url(@book), notice: 'Book was successfully created.' }
```
遅延探索を活用できていることは良いのですが、本以外のコントローラが追加されたときにも類似したメッセージ( `Book` の部分だけを別のモデル名で置き換えたもの)を定義する必要が出てしまいます。

ここは他のコントローラでも使えるようキー設計を見直して、かつモデル名の部分を置き換えられるようにしましょう。

a-kuroki-gs replied:
遅延探索を使わずにcontrollerの共通部分として`success`等を定義しました。
en.ymlとja.ymlでは`book`と明記せずに`%{model}`と変更しております。

yuuu commented on config/application.rb:
```diff
@@ -10,6 +10,8 @@ module BooksApp
   class Application < Rails::Application
     # Initialize configuration defaults for originally generated Rails version.
     config.load_defaults 7.0
+    config.i18n.load_path += Dir[Rails.root.join('my', 'locales', '*.{rb, yml}')]
```
こちらの記述は必要でしょうか？

a-kuroki-gs replied:
config / locales内にロケールファイルをすべて置いているため、特に指定する必要はありませんでした。
削除しております。

yuuu commented on config/locales/ja.yml:
```diff
@@ -0,0 +1,183 @@
+  activerecord:
+    errors:
+      messages:
+        record_invalid: 'バリデーションに失敗しました: %{errors}'
```
このあたりの翻訳ももれなく定義されていて素晴らしいです ✨ ただ、 [rails-i18n](https://github.com/svenfuchs/rails-i18n) というgemを使うと、Rails自体が提供している翻訳は自動的に日本語化されるはずですので、活用を検討ください。

a-kuroki-gs replied:
`rails-i18n`gemを導入し、ロケールファイルとビューファイルを一部変更しました。

yuuu commented on config/locales/translation_ja.yml:
```diff
@@ -0,0 +1,45 @@
+ja:
```
rails-i18nを活用する前提であれば、このファイルを `ja.yml` にリネームできそうですね。

a-kuroki-gs replied:
`translation_ja.yml`ファイルを`ja.yml`にリネームしました。

a-kuroki-gs committed:
他のコントローラでも使えるようキー設計を見直しました

books_controller.rbで、遅延探索を使わずにcontrollerの共通部分として等を定義
en.ymlとja.ymlではbookと明記せずに%{model}として汎用性を上げる

a-kuroki-gs committed:
application.rbの不要な文を削除

現状、config > locales内にロケールファイルをすべて置いているから、特に指定する必要はない
よくわからずに記述していたが不要だったため削除

a-kuroki-gs committed:
rails-i18n gemを導入しました

a-kuroki-gs committed:
ロケールファイルを変更しました

エラーメッセージを自分で作成していたが、gemで翻訳できるため削除
ビューファイルはエラーメッセージのタイトルのところを翻訳できるように変更
その他不要な記述があったので削除
また、ja.ymlファイルとen.ymlファイルに必要な記述を残し、translation_ja.ymlは削除
